### ISSUE TITLE
Opencost returns 0 in node_gpu_hourly_cost metric for new GCP machine families with GPUs

### ISSUE THREAD

**Issue by KKatsnel** [opencost/opencost#2215]
Opencost returns 0 in node_gpu_hourly_cost metric for new GCP machine families with Nvidia A100 80 Gb, Nvidia L4, and Nvidia H100. For example - for a2-ultragpu-1g, g2-standard-4, a3-highgpu-8g instances.

1. Add an a2-ultragpu-1g instance as a node to a GKE cluster
2. Get the node_gpu_hourly_cost metric
3. Instead of the actual price, there will be "0" for the node

Opencost 1.106.2.

**Comment by pawel-lmcb** [comment]
I'm wondering whether it's related to https://github.com/opencost/opencost/issues/2029 putting this here just incase

**Comment by mattray** [comment]
@KKatsnel thanks for raising this. I'll refer it over to Kubecost Engineering to see if they can investigate.

**Comment by KKatsnel** [comment]
As far as I see, it is a pure Opencost issue, for example a2-ultragpu machines must be added here: https://github.com/opencost/opencost/blob/64eaa8c677e9659dfe52f4f6140a379bd99dfb90/pkg/cloud/gcp/provider.go#L794

**Comment by AjayTripathy** [comment]
Yep-- looks like it. We'll get this prioritized and added.

**Comment by dwbrown2** [comment]
Good to close after saweber's work?

**Comment by KKatsnel** [comment]
Unfortunately no, there are no g2-standard-* and a3-highgpu-* machines in the corresponding commit.

**Comment by cliffcolvin** [comment]
@KKatsnel will get these other two instance types added.

**Comment by saweber** [comment]
I apologize for misreading the issue; I overlooked g2 and a3 in the description.

The above PR should catch g2 instances now; TL;DR - our logic only handled situations where the GPU was an 'Nvidia Tesla', and not just 'Nvidia'. This is fixed.

Currently, Kubecost) does not have access to the a3 instance family, nor has GCP publicly published detailed information yet. I have filed an internal infrastructure ticket (IN-121) to ask for access, but cannot provide a timeline for when we can implement a fix for the a3 family.
